src/search3_response.erl

```diff
@@ -65,6 +66,15 @@ order_to_json(Order) ->
     end,
     lists:map(OrderFun, Order).
 
+facets_to_json(Counts) ->
+    CountsList = maps:to_list(Counts),
+    CountsFun = fun
+        ({K, V}) when is_binary(K), is_map(V) ->
+            #{counts := SubCounts} = V,
+            SubCountsList = maps:to_list(SubCounts),
+            {K, {SubCountsList}}
+    end,
+    {lists:map(CountsFun, CountsList)}.
```

Comment:
not sure why we turn the map into a list when jiffy can handle maps?

```
jiffy:encode(#{<<"foo">> => <<"bar">>}).
<<"{\"foo\":\"bar\"}">>
```

Reply:
@rnewson: If we did a jiffy:encode, the binary format
`<<"{\"class\":{\"counts\":{\"mammal\":8,\"bird\":2}}}">>` is returned to the user in readable format but it breaks existing testy tests that expects a nested object. 

The list conversion creates the following nested object:
```
    "counts": {
        "class": {
            "bird": 2,
            "mammal": 8
        }
    }
```

Reply:
fair enough.